app/controllers/BannerTestsController.scala

```diff
@@ -0,0 +1,128 @@
+package controllers
+
+import com.gu.googleauth.AuthAction
+import play.api.mvc.{AnyContent, ControllerComponents, Result}
+import models.{BannerTest, BannerTests}
+import BannerTestsController._
+import play.api.libs.ws.WSClient
+import services.{FastlyPurger, S3Json, VersionedS3Data}
+import services.S3Client.S3ObjectSettings
+import io.circe.generic.auto._
+import io.circe.syntax._
+import play.api.libs.circe.Circe
+import zio.blocking.Blocking
+import zio.{DefaultRuntime, IO, ZIO}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object BannerTestsController {
```

Comment:
pretty much all of the archive logic here is the same as in the EpicTestsController.
Can we make it a trait, along with the `run` method?
That way `EpicTestsController` and `BannerTestsController` would have nothing in their bodies

Reply:
@tjsilver shall we have a look at this when you're back ?

Reply:
This turned out to be quite hard because:
1. it cannot be a class, because we cannot mix in more than 1 class,
1. traits cannot have generic parameters with context bounds

Let's look at it another time...